src/cds/record/module.js

```diff
@@ -122,6 +122,36 @@
       vm.cdsRecordWarning = warning;
     }
 
+    $scope.logMediaDownload = function (fileObj) {
+      if (!$scope.mediaDownloadEventUrl) {
+        return;
+      }
+
+      var quality = fileObj.context_type === 'master' ? 'master' : fileObj.tags.preset_quality,
+          replacedUrl = replaceUrlParams($scope.mediaDownloadEventUrl, $scope.record.metadata, fileObj.key, quality);
+
+      $http.get(replacedUrl)
+        .then(function(response) {
+        })
+        .then(function(error) {
+        });
+
+      function replaceUrlParams(url, recordMetadata, filename, quality) {
```

Comment:
just a question do we need this function inside ``logMediaDownload``? :)

Reply:
Not mandatory, it is done for readability/encapsulation/responsibilities.
What do you propose to improve the code?

Reply:
:/ it's just for each time we call ``logMediaDownload``, ``replaceUrlParams`` will initialize again, WDTY about moving it outside?

Reply:
Done!
I like it less in term of function visibility and encapsulation, but small details...